spec/acceptance/api/internal/users_api_spec.rb

```diff
@@ -144,4 +144,14 @@
     end
   end
 
+  delete '/services/:service_id/users/all' do
+    parameter :service_id, 'Service ID', required: true
+
+    example_request 'Deleting all Users of a Service' do
+      expect(status).to eq 200
+      expect(response_json['status']).to eq 'deleted'
+      expect(ThreeScale::Backend::User.delete_all(service_id)).to be nil
```

Comment:
I think that this `expect` assumes things that are not documented nor tested.

In the unit tests of `User` you did not check the return value of `delete_all`, you just verified that after calling it, users no longer exists. I think that's fine, but the here we should just do the same.

The alternative would be to define a return value for `delete_all` in docs/tests, but I don't think it's needed in this case.
What do you think?

Reply:
I now check if the user has been deleted or not instead of the return value of the delete_all method